### PR TITLE
fix: EPMLSTRCMW-180 User can create invalid password

### PR DESCRIPTION
**Pull request by eshenko**

### CODE REVIEW

**Review by myazinn — COMMENTED** [review]
Regex changes look good to me, but why do you need to change HTTP method from `put` to `post`?

**Comment by eshenko** [comment]
> Regex changes look good to me, but why do you need to change HTTP method from `put` to `post`?

There were two equal Put endpoints in the UserController. For updating user information and for changing user's password. When we send an incorrect password, we go into the updating user information endpoint, instead of going to the changing user's password endpoint.

I had two options:
1. To add some path (for example, users/info and users/password)
2. To change the method of one of the endpoints

I chose the second one. Am I wrong?

**Comment by eshenko** [comment]
Fixed.
Added paths `users/info` and `users/password` in the Put methods of the UserController.

File: controllers/src/main/scala/cromwell/pipeline/controller/UserController.scala [hunk 1]
```diff
@@ -14,32 +14,38 @@ import scala.util.{ Failure, Success }
...
-    path("users") {
-      concat(
-        get {
-          parameter('email.as[String]) { email =>
-            onComplete(userService.getUsersByEmail(email)) {
-              case Success(r) => complete(r)
-              case Failure(exc) =>
-                complete(StatusCodes.InternalServerError, exc.getMessage)
+    concat(
+      path("users") {
```
Comment (myazinn) [inline controllers/src/main/scala/cromwell/pipeline/controller/UserController.scala]: Let's make it DRY. You don't need concatenation here and can reuse `path("users")` for `put` endpoints
```
path(users) {
  concat (
    get { },
    delete { },
    path(info) { put { } },
    path(password) { put { } }
  )
}
```
or
```
path(users) {
  concat (
    get { },
    delete { },
    put {
      concat(
        path(info) { },
        path(password) { }  
      )
    }
  )
}
```
Which one to choose is up to you.

Comment (eshenko) [reply controllers/src/main/scala/cromwell/pipeline/controller/UserController.scala]: fixed.

**Review by GrigoriiBerezin — APPROVED** [review]
Nice

**Review by myazinn — APPROVED** [review]

**Pull request merged**